librz/core/tui/panels.c

```diff
@@ -4276,20 +4284,82 @@ void __del_menu(RzCore *core) {
 	menu->n_refresh = menu->depth - 1;
 }
 
+/**
+ * \brief Create a new menu item with optionals name \p name and callback \p cb
+ */
+RZ_OWN RzPanelsMenuItem *rz_panels_menu_item_new(RZ_NULLABLE const char *name, RZ_NULLABLE RzPanelsMenuCallback cb) {
+	RzPanelsMenuItem *item = RZ_NEW0(RzPanelsMenuItem);
+	if (!item) {
+		return NULL;
+	}
+	item->selectedIndex = 0;
+	rz_pvector_init(&item->sub_vec, (RzPVectorFree)rz_panels_menu_item_free);
+	item->name = name ? strdup(name) : NULL;
+	item->cb = cb;
+	return item;
+}
+
+void rz_panels_menu_item_free(RZ_NULLABLE RzPanelsMenuItem *item) {
+	if (!item) {
+		return;
+	}
+	free(item->name);
+	rz_panel_free(item->p);
+	rz_pvector_fini(&item->sub_vec);
+	free(item);
+}
+
+/**
+ * \brief Select next sub-menu
+ */
+void panels_menu_item_next_sub(RZ_NONNULL RzPanelsMenuItem *item) {
+	rz_return_if_fail(item);
+
+	item->selectedIndex++;
+	if (item->selectedIndex == rz_pvector_len(&item->sub_vec)) {
```

Comment:
i have never been a fan of `==` for these usages. i always recommand to use `>=` because the edge case is always any number that is greater or equal to `len`.